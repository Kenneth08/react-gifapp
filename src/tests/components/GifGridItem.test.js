import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe("Test into GifGridItem.js", () => {
  const title = "Is a title.";
  const url = "https://localhost.image.jpg";

  let wrapper = shallow(<GifGridItem title={title} url={url} />);
  beforeEach(() => {
    wrapper = shallow(<GifGridItem title={title} url={url} />);
  });
  test("should show the component successfully", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("should get title value", () => {
    const p = wrapper.find("p").text().trim();
    expect(p).toBe(title);
  });
  test("should get img props", () => {
    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });
  test('should get className "animate__fadeIn"', () => {
    const div = wrapper.find("div");
    const className = div.prop("className");
    expect(className.includes("animate__fadeIn")).toBe(true);
  });
});
