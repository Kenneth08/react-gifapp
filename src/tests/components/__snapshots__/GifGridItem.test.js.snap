// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test into GifGridItem.js should show the component successfully 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Is a title."
    src="https://localhost.image.jpg"
  />
  <p>
    Is a title.
  </p>
</div>
`;
